Automattic/wp-calypso
# Hosting Command Palette: add tracks navigation to wp-admin and refactor navigation

- Related to https://github.com/Automattic/dotcom-forge/issues/4670

## Proposed Changes

* Simplify onClick function types
* Refactor navigations
* Add navigation track event `calypso_hosting_command_palette_navigate`
  * Example: 
  * `command : "openSiteDashboard"`
  * `current_route : "/hosting-config/:site"`
  * `is_wp_admin : false`

## Testing Instructions



* Enable event logs in Chrome dev tools: `localStorage.setItem( 'debug', 'calypso:analytics*' );` or use the vigilante chrome extension
* Open the command palette by pressing `cmd+k`.
* Select a command to navigate inside Calypso , like `Open site dashboard`
* Observe the event `calypso_hosting_command_palette_navigate` is fired with the correct properties. `is_wp_admin: false`
* Select a command to navigate to wp-admin , like `Manage plugins` on a Jetpack site or atomic with the classic admin interface enabled.
* Observe the event `calypso_hosting_command_palette_navigate` is fired with the correct properties. `is_wp_admin: true`

### Screencast


https://github.com/Automattic/wp-calypso/assets/779993/c837c089-f3c1-42ca-9720-d8ecb6951c29

<img width="1908" alt="Screenshot 2023-12-21 at 02 38 03" src="https://github.com/Automattic/wp-calypso/assets/779993/97162314-1696-4c1f-9dd8-7cb19dcc67dc">

<img width="1838" alt="Screenshot 2023-12-21 at 02 38 28" src="https://github.com/Automattic/wp-calypso/assets/779993/8f42202b-de35-4818-82af-303185bb9dee">


## Pre-merge Checklist
